service/topology_coordinator.cc

```diff
@@ -1617,11 +1623,10 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
                 on_internal_error(rtlogger, "should_preempt_balancing() retook the guard");
             }
         }
-        if (!preempt) {
-            auto plan = co_await _tablet_allocator.balance_tablets(get_token_metadata_ptr(), _tablet_load_stats, get_dead_nodes());
-            if (!drain || plan.has_nodes_to_drain()) {
-                co_await generate_migration_updates(updates, guard, plan);
-            }
+
+        auto plan = co_await _tablet_allocator.balance_tablets(get_token_metadata_ptr(), _tablet_load_stats, get_dead_nodes());
+        if (!preempt && (!drain || plan.has_nodes_to_drain())) {
```

Comment:
No need for this change after addressing my comment https://github.com/scylladb/scylladb/pull/22148/files#r1901913770

Reply:
ack